src/main/java/duke/command/MainUI.java

```diff
@@ -0,0 +1,149 @@
+            String taskSentence = sc.nextLine();
+            char taskCategory = taskSentence.charAt(1);
+            boolean isDone = taskSentence.substring(4,5).compareTo("\u2713") == 0;
+            switch (taskCategory) {
+            case 'T':
+                String toDoDescription = taskSentence.substring(7);
+                Todo t = new Todo(toDoDescription);
+                if (isDone){
+                    t.markAsDone();
+                }
+                taskArrayList.add(t);
+                break;
+            case 'D':
+                int dueDateStartingIndex = taskSentence.indexOf("(by:");
+                String dueDate = taskSentence.substring(dueDateStartingIndex + 5, taskSentence.length()-1);
+                String deadlineDescription = taskSentence.substring(7,dueDateStartingIndex);
+//                System.out.println(dueDateStartingIndex+":"+dueDate);
+                Deadline d = new Deadline(deadlineDescription,dueDate);
+                if (isDone){
+                    d.markAsDone();
+                }
+                taskArrayList.add(d);
+                break;
+            case 'E':
+                int durationStartingIndex = taskSentence.indexOf("(at:");
+                String duration = taskSentence.substring(durationStartingIndex + 5, taskSentence.length()-1);
+                String eventDescription = taskSentence.substring(7,durationStartingIndex);
+//                System.out.println(durationStartingIndex+":"+duration);
+                Event e = new Event(eventDescription,duration);
+                if (isDone){
+                    e.markAsDone();
+                }
+                taskArrayList.add(e);
+                break;
+            default:
+                break;
+            }
+        }
+        System.out.println("Done loading the data!");
+        printDivider();
+
+    }
+
+    private static void runProgram(Scanner in) {
+        try {
+            File data = new File("data/duke.txt");
+            if (data.createNewFile()) {
+                System.out.println("Your file that stores all your tasks is not found! (data/duke.txt) :(");
+                System.out.println("Creating new file now to store your future tasks! :)");
+                System.out.println("File created: " + data.getName());
+                printDivider();
+            } else {
+                loadData();
+            }
+
+        } catch (IOException e) {
+            System.out.println("We couldn't find the 'data' folder in your directory. This data folder will store your tasks.");
+            System.out.println("Creating 'data' folder now...");
+            File relativePathFileName = new File(DIRECTORY_NAME);
+            String fullPath = relativePathFileName.getAbsolutePath();
+            File absolutePathFileName = new File(fullPath);
+            boolean bool = absolutePathFileName.mkdirs();
+            if(bool) {
+                System.out.println("'data' folder created successfully");
```

Comment:
You could consider refactoring this code as your method is a little long. See [avoid long methods](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods).

Reply:
Thanks! I have made the changes :)